app/src/main/java/com/weatherxm/util/NumberUtils.kt

```diff
@@ -1,35 +1,61 @@
 package com.weatherxm.util
 
 import android.icu.text.CompactDecimalFormat
-import android.icu.text.NumberFormat
-import com.weatherxm.util.Weather.EMPTY_VALUE
+import com.weatherxm.data.DECIMAL_FORMAT_TOKENS
+import com.weatherxm.ui.common.Contracts.EMPTY_VALUE
 import org.koin.core.component.KoinComponent
 import org.koin.core.component.inject
+import org.koin.core.qualifier.named
+import java.math.BigDecimal
 import java.math.RoundingMode
+import java.text.DecimalFormat
+import java.text.NumberFormat
 
 object NumberUtils : KoinComponent {
     private val compactDecimalFormat: CompactDecimalFormat by inject()
     private val numberFormat: NumberFormat by inject()
+    private val decimalFormat: DecimalFormat by inject(named(DECIMAL_FORMAT_TOKENS))
+
+    private const val DIVISOR_WEI_TO_ETH = "1000000000000000000"
+    private const val ETH_DECIMALS = 18
 
     fun compactNumber(number: Number?): String {
         return number?.let {
             compactDecimalFormat.format(number)
         } ?: EMPTY_VALUE
     }
 
-    fun formatNumber(number: Number?, maxDecimals: Int = 0): String {
+    fun formatNumber(number: Number?, decimals: Int = 0): String {
         return number?.let {
-            numberFormat.maximumFractionDigits = maxDecimals
+            numberFormat.minimumFractionDigits = decimals
+            numberFormat.maximumFractionDigits = decimals
             numberFormat.format(it)
         } ?: EMPTY_VALUE
     }
 
-    fun roundToDecimals(value: Number, decimals: Int = 1): Float {
-        return value.toFloat().toBigDecimal().setScale(decimals, RoundingMode.HALF_UP).toFloat()
+    fun formatTokens(amount: Float?): String {
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/WeatherXM/wxm-android/pull/342#issuecomment-2388448877>#04ac52</a><span> - 10/02/2024, 12:12 pm</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The formatTokens function for Float may introduce rounding errors when converting to BigDecimal, which could lead to inaccuracies in financial calculations.
</div>
<div id="fix">
<b>Fix:</b> Use BigDecimal for the input parameter instead of Float to maintain precision. If Float must be used, consider using BigDecimal's valueOf method for more accurate conversion.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -36,7 +36,7 @@ object NumberUtils : KoinComponent {
 -    fun formatTokens(amount: Float?): String {
 -        return amount?.let {
 -            formatTokens(it.toBigDecimal())
 +    fun formatTokens(amount: Float?): String {
 +        return amount?.let {
 +            formatTokens(BigDecimal.valueOf(it.toDouble()))
      } ?: EMPTY_VALUE
  }
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged